pkg/controller/daemon/update.go

```diff
@@ -570,3 +614,35 @@ func (h historiesByRevision) Swap(i, j int) { h[i], h[j] = h[j], h[i] }
 func (h historiesByRevision) Less(i, j int) bool {
 	return h[i].Revision < h[j].Revision
 }
+
+func (dsc *DaemonSetsController) checkShouldDeleteDaemonPod(nodeName string, ds *apps.DaemonSet) (bool, error) {
```

Comment:
can we first try to do without this function and have the code duplicated first?  At this point it seems like a needless misdirection.

Reply:
OK, I've fixed that.